src/actions/MessageCenterActions.ts

```diff
@@ -23,56 +23,27 @@ const toNumberArray = (n: numberOrNumberArray) => (Array.isArray(n) ? n : [n]);
 export const MessageCenterActions = {
   addMessage: createAction(
     MessageCenterActionKeys.ADD_MESSAGE,
-    (content: string, groupId: string = DEFAULT_GROUP_ID, messageType: MessageType = DEFAULT_MESSAGE_TYPE) => ({
-      type: MessageCenterActionKeys.ADD_MESSAGE,
-      content,
-      groupId,
-      messageType
-    })
+    resolve => (content: string, groupId: string = DEFAULT_GROUP_ID, messageType: MessageType = DEFAULT_MESSAGE_TYPE) =>
+      resolve({ content, groupId, messageType })
+  ),
+  removeMessage: createAction(MessageCenterActionKeys.REMOVE_MESSAGE, resolve => (messageId: numberOrNumberArray) =>
+    resolve({ messageId: toNumberArray(messageId) })
+  ),
+  markAsRead: createAction(MessageCenterActionKeys.MARK_MESSAGE_AS_READ, resolve => (messageId: numberOrNumberArray) =>
+    resolve({ messageId: toNumberArray(messageId) })
+  ),
+  toggleGroup: createAction(MessageCenterActionKeys.TOGGLE_GROUP, resolve => (groupId: string) => resolve({ groupId })),
+  expandGroup: createAction(MessageCenterActionKeys.EXPAND_GROUP, resolve => (groupId: string) => resolve({ groupId })),
+  hideNotification: createAction(
+    MessageCenterActionKeys.HIDE_NOTIFICATION,
+    resolve => (messageId: numberOrNumberArray) => resolve({ messageId: toNumberArray(messageId) })
   ),
-  removeMessage: createAction(MessageCenterActionKeys.REMOVE_MESSAGE, (messageId: numberOrNumberArray) => {
-    const type = MessageCenterActionKeys.REMOVE_MESSAGE;
-    messageId = toNumberArray(messageId);
-    return {
-      type,
-      messageId
-    };
-  }),
-  markAsRead: createAction(MessageCenterActionKeys.MARK_MESSAGE_AS_READ, (messageId: numberOrNumberArray) => {
-    const type = MessageCenterActionKeys.MARK_MESSAGE_AS_READ;
-    messageId = toNumberArray(messageId);
-    return {
-      type,
-      messageId
-    };
-  }),
-  toggleGroup: createAction(MessageCenterActionKeys.TOGGLE_GROUP, (groupId: string) => {
-    const type = MessageCenterActionKeys.TOGGLE_GROUP;
-    return {
-      type,
-      groupId
-    };
-  }),
-  expandGroup: createAction(MessageCenterActionKeys.EXPAND_GROUP, (groupId: string) => {
-    const type = MessageCenterActionKeys.EXPAND_GROUP;
-    return {
-      type,
-      groupId
-    };
-  }),
-  hideNotification: createAction(MessageCenterActionKeys.HIDE_NOTIFICATION, (messageId: numberOrNumberArray) => {
-    const type = MessageCenterActionKeys.HIDE_NOTIFICATION;
-    messageId = toNumberArray(messageId);
-    return {
-      type,
-      messageId
-    };
-  }),
-
   showMessageCenter: createAction(MessageCenterActionKeys.SHOW),
   hideMessageCenter: createAction(MessageCenterActionKeys.HIDE),
-  togleExpandedMessageCenter: createAction(MessageCenterActionKeys.TOGGLE_EXPAND),
+  togleExpandedMessageCenter: createAction(MessageCenterActionKeys.TOGGLE_EXPAND)
```

Comment:
I don't understand most of what's in this PR :)  But I do know how to spell. That should be 'toggle' not 'togle'.  I know, super helpful...

Reply:
Good spelling also fosters good quality code ;)
I'll fix.